contracts/StakingProxy.cdc

```diff
@@ -0,0 +1,156 @@
+// This contract defines an interface for node stakers
+// to use to be able to perform common staking actions
+
+// It also defines a resource that a node operator can
+// use to store staking proxies for all of their node operation
+// relationships 
+
+pub contract StakingProxy {
+
+    /// path to store the node operator resource
+    /// in the node operators account for staking helper
+    pub let NodeOperatorCapabilityStoragePath: Path
+
+    pub let NodeOperatorCapabilityPublicPath: Path
+
+    /// Contains the node info associated with a node operator
+    pub struct NodeInfo {
+
+        pub let id: String
+        pub let role: UInt8
+        pub let networkingAddress: String
+        pub let networkingKey: String
+        pub let stakingKey: String
+
+        init(nodeID: String, role: UInt8, networkingAddress: String, networkingKey: String, stakingKey: String) {
+            pre {
+                networkingAddress.length > 0 && networkingKey.length > 0 && stakingKey.length > 0:
+                        "Address and Key have to be the correct length"
+            }
+            self.id = nodeID
+            self.role = role
+            self.networkingAddress = networkingAddress
+            self.networkingKey = networkingKey
+            self.stakingKey = stakingKey
+        }
+    }
+
+    /// The interface that limits what a node operator can access
+    /// from the staker who they operate for
+    pub struct interface NodeStakerProxy {
+
+        pub fun stakeNewTokens(amount: UFix64)
+
+        pub fun stakeUnlockedTokens(amount: UFix64)
+
+        pub fun requestUnstaking(amount: UFix64)
+
+        pub fun unstakeAll()
+
+        pub fun withdrawUnlockedTokens(amount: UFix64)
+
+        pub fun withdrawRewardedTokens(amount: UFix64)
+
+    }
+
+    /// The interface the describes what a delegator can do
+    pub struct interface NodeDelegatorProxy {
+
+        pub fun delegateNewTokens(amount: UFix64)
+
+        pub fun delegateUnlockedTokens(amount: UFix64)
+        
+        pub fun delegateRewardedTokens(amount: UFix64)
+
+        pub fun requestUnstaking(amount: UFix64)
+
+        pub fun withdrawUnlockedTokens(amount: UFix64)
+
+        pub fun withdrawRewardedTokens(amount: UFix64)
+    }
+
+    /// The interface that a node operator publishes their NodeStakerProxyHolder
+    /// as in order to allow other token holders to initialize
+    /// staking helper relationships with them
+    pub resource interface NodeStakerProxyHolderPublic {
+        
+        pub fun addStakingProxy(nodeID: String, proxy: AnyStruct{NodeStakerProxy})
+
+        pub fun getNodeInfo(nodeID: String): NodeInfo?
+    }
+
+    /// The resource that node operators store in their accounts
+    /// to manage relationships with token holders who pay them off-chain
+    /// instead of with tokens
+    pub resource NodeStakerProxyHolder: NodeStakerProxyHolderPublic {
```

Comment:
```suggestion
    pub resource NodeStakerProxyContainer: NodeStakerProxyHolderPublic {
```

Maybe this should be "container" to be consistent with other resources and to avoid confusion with the _token_ holder?